torchdynamo/guards.py

```diff
@@ -274,20 +382,20 @@ def TUPLE_ITERATOR_LEN(self, guard):
         self.code.append(f"___tuple_iterator_len({ref}) == {tuple_iterator_len(value)}")
 
     def DICT_KEYS(self, guard):
+        if guard.is_nn_module():
+            # Protected by module invalidation, see NN_MODULE
+            return
```

Comment:
I think this will be incorrect:
```
class Foo(nn.Module):
   def __init__(self):
        super().__init__()
        self.my_dict = collection.OrderedDict()

   def forward(self, x):
        if "x" in self.my_dict:
              return x + 1
        else:
              return x - 1

m = Foo()
with torchdynamo.optimize("eager"):
   m(torch.randn(10))
   m.my_dict["x"] = 1
   m(torch.randn(10))
```